app/models/vendor.rb

```diff
@@ -34,6 +34,20 @@ def self.accessible_by(user)
     end
   end
 
+  # This method does nothing more than attempt to cleanup a lot of data instead of making rails do it,
+  # since rails is really bad at cleaning up quickly. Note that this bypasses the dependent => destroy
+  # calls on product, product tests, etc, all the way down and does them manually. This means that
+  # if any of those structures change then this code will need to be updated accordingly.
+  def destroy
+    product_tests = ProductTest.where(:product_id.in => product_ids)
+    product_test_ids = product_tests.pluck(:_id)
+    ProductTest.destroy_by_ids(product_test_ids)
+
+    Product.in(id: product_ids).delete
+
+    super
```

Comment:
Correct, the call to super doesn't slow anything down and it still calls the before_destroy and after_destroy methods for the vendor model, and will delete any dependent associations that were missed somehow.